### ISSUE TITLE
【测试用例】评测任务模块重构 E2E 测试用例更新规划（#103 #98 #99）

### ISSUE THREAD

**Issue by chenxibj** [chenxibj/ai-hardware-verification-platform#104]
## 背景

评测任务模块正在进行重大重构（#103 #98 #99），现有 E2E 测试用例需要对应更新。本 issue 规划重构后需要新增和修改的所有 E2E 测试用例。

**等重构代码合入后再实现，本 issue 作为测试用例规划清单。**

## 现有测试覆盖范围（9 个 spec 文件）

| 文件 | 覆盖范围 | 受重构影响 |
|------|---------|-----------|
| `task-crud.spec.ts` | 任务创建向导（4步）、自定义/模板模式、列表展示、搜索 | ⚠️ **重度影响** — 创建流程完全重构 |
| `task-lifecycle.spec.ts` | 创建→初始状态、取消、详情查看、删除、克隆 | ⚠️ **重度影响** — 状态机扩展、新操作 |
| `batch-operations.spec.ts` | 批量选择、批量取消/删除 | 🔶 中度影响 — 可能需要适配新状态 |
| `data-consistency.spec.ts` | 创建后列表一致性、统计更新、删除一致性 | 🔶 中度影响 — 创建流程变化 |
| `smoke.spec.ts` | 全页面冒烟测试 | ✅ 低影响 |
| `error-handling.spec.ts` | 登录校验、空搜索、表单验证 | 🔶 中度影响 — 表单验证逻辑变化 |
| `workflow.spec.ts` | 工作流编排 | ✅ 不受影响 |
| `resource.spec.ts` | 资源管理 | ✅ 不受影响 |
| `report.spec.ts` | 评测报告 | ✅ 不受影响 |

### `tasks.page.ts` 需要重构

**需要新增/修改的属性和方法：**

```
// 新增属性
evalDimensionSelect    // 评测维度下拉（主驱动字段）
evalObjectPicker       // 评测对象选择器（从对象库选择，非自由文本）
evalObjectEmptyTip     // 对象库为空时的提示文案
tagInput               // 标签输入
presetTaskBadge        // 系统预置标记
cloneButton            // 克隆按钮
pauseButton            // 暂停按钮
resumeButton           // 恢复按钮
restartButton          // 重启按钮

// 新增方法
selectEvalDimension(dimension: string)      // 选择评测维度
selectEvalObject(objectName: string)        // 从对象库选择评测对象
selectMultipleEvalObjects(names: string[])  // 多选评测对象（算子维度）
getEvalTypeOptions(): Promise<string[]>     // 获取当前可选的评测类型列表
isEvalTypeLocked(): Promise<boolean>        // 判断评测类型是否自动锁定
pauseTask(index: number)                    // 暂停任务
resumeTask(index: number)                   // 恢复任务
restartTask(index: number)                  // 重启任务
cloneTask(index: number)                    // 克隆任务
getActionButtons(index: number): Promise<string[]>  // 获取某行可见的操作按钮列表
```

**需要修改的方法：**
- `fillBasicInfo()` — 参数结构变化：去掉 `evalObject` 自由文本，改为 `evalDimension` + 从对象库选择
- `getTaskStatus()` — 适配新增的状态（已暂停等）

---

### 1.1 评测维度×评测类型联动（核心）

> 文件：`task-crud.spec.ts` 新增/修改

| # | 用例名称 | 步骤 | 预期 |
|---|---------|------|------|
| TC-103-01 | 算子维度→类型自动锁定为性能评测 | 选择评测维度=算子评测 | 评测类型自动选中"性能评测"，且不可编辑（只读/disabled） |
| TC-103-02 | 芯片维度→类型自动锁定为性能评测 | 选择评测维度=芯片评测 | 评测类型自动选中"性能评测"，且不可编辑 |
| TC-103-03 | 框架维度→类型自动锁定为兼容性评测 | 选择评测维度=框架评测 | 评测类型自动选中"兼容性评测"，且不可编辑 |
| TC-103-04 | 模型维度→类型可选性能/精度 | 选择评测维度=模型评测 | 评测类型下拉可选"性能评测"和"精度评测"两项 |
| TC-103-05 | 中间层维度→类型可选性能/兼容性 | 选择评测维度=中间层评测 | 评测类型下拉可选"性能评测"和"兼容性评测"两项 |
| TC-103-06 | 场景维度→类型可选性能/精度/稳定性 | 选择评测维度=场景评测 | 评测类型下拉可选"性能评测""精度评测""稳定性评测"三项 |

### 1.2 切换维度清空联动

| # | 用例名称 | 步骤 | 预期 |
|---|---------|------|------|
| TC-103-07 | 切换维度清空评测类型 | 选择模型维度→选中精度评测→切换为算子维度 | 评测类型被清空后自动锁定为性能评测 |
| TC-103-08 | 切换维度清空评测对象 | 选择算子维度→选择若干算子→切换为芯片维度 | 评测对象被清空，显示芯片列表 |

### 1.3 评测对象从对象库选择

| # | 用例名称 | 步骤 | 预期 |
|---|---------|------|------|
| TC-103-09 | 算子维度对象库多选 | 选择算子评测→点击评测对象选择器 | 弹出对象选择列表，显示已注册算子，支持多选 |
| TC-103-10 | 芯片/模型/框架维度对象库单选 | 选择芯片评测→点击评测对象选择器 | 弹出对象选择列表，显示已注册芯片，单选模式 |
| TC-103-11 | 对象库为空时提示 | 选择某维度，该维度对象库无数据 | 显示"请先在评测对象管理中注册"提示，且有跳转链接 |
| TC-103-12 | 对象库数据实时加载 | 切换维度后 | 对象选择器调用 API 拉取对应类别对象列表 |

### 1.4 表单校验与无效组合拦截

| # | 用例名称 | 步骤 | 预期 |
|---|---------|------|------|
| TC-103-13 | 未选维度不能进入下一步 | 不选评测维度，直接点下一步 | 显示校验错误，阻止跳转 |
| TC-103-14 | 未选评测对象不能提交 | 选维度和类型，不选对象，尝试提交 | 显示校验错误，阻止提交 |
| TC-103-15 | 必填字段为空时阻止提交 | 任务名称为空，尝试提交 | 显示校验错误 |

### 1.5 完整创建流程（替换旧用例）

| # | 用例名称 | 步骤 | 预期 |
|---|---------|------|------|
| TC-103-16 | 自定义模式完整创建（算子维度） | 选自定义→算子维度→性能评测（自动锁定）→多选算子→提交 | 创建成功，列表展示维度="算子评测"，对象="MatMul, Conv2D..." |
| TC-103-17 | 自定义模式完整创建（模型维度） | 选自定义→模型维度→选精度评测→选模型→提交 | 创建成功 |
| TC-103-18 | 模板模式创建 | 选模板→芯片性能评测模板→验证维度/类型预填→提交 | 创建成功，预填字段正确 |

### 1.6 列表页展示

| # | 用例名称 | 步骤 | 预期 |
|---|---------|------|------|
| TC-103-19 | 列表页显示评测维度列 | 查看任务列表 | 表格包含"评测维度"列，显示"算子评测/模型评测"等 |
| TC-103-20 | 列表页显示评测对象列 | 查看任务列表 | 表格包含"评测对象"列，显示具体对象名称（非类型） |
| TC-103-21 | 按评测维度筛选列表 | 使用维度筛选器选择"算子评测" | 列表仅显示算子评测维度的任务 |

---

## 二、#98 算子评测预置任务 — 测试用例

> 新文件：`preset-task.spec.ts`

| # | 用例名称 | 步骤 | 预期 |
|---|---------|------|------|
| TC-098-01 | 新用户可见预置任务 | 登录后进入评测任务列表 | 列表中存在标记为「系统预置」的示例评测任务 |
| TC-098-02 | 预置任务标记展示 | 查看预置任务行 | 带有「系统预置」标签/Badge，区分于用户创建任务 |
| TC-098-03 | 预置任务包含关键算子 | 打开预置任务详情 | 评测对象包含 MatMul、Conv2D、ReLU、Softmax、BatchNorm、Pooling |
| TC-098-04 | 预置任务不可编辑 | 点击预置任务的编辑/修改按钮 | 编辑按钮不可见或禁用，提示"系统预置任务不可修改" |
| TC-098-05 | 预置任务可直接运行 | 点击预置任务的运行按钮 | 任务开始执行，状态变为运行中 |
| TC-098-06 | 预置任务支持克隆 | 点击预置任务的克隆按钮 | 成功创建副本，副本可编辑 |
| TC-098-07 | 克隆后的任务配置正确 | 克隆预置任务后查看详情 | 所有评测配置（维度、类型、对象、参数）与原任务一致 |
| TC-098-08 | 预置任务不可删除 | 尝试删除预置任务 | 删除按钮不可见或禁用 |

---

## 三、#99 评测任务生命周期管理 — 测试用例

> 文件：`task-lifecycle.spec.ts` 重构

### 3.1 操作按钮动态显示

| # | 用例名称 | 步骤 | 预期 |
|---|---------|------|------|
| TC-099-01 | 待运行状态→可见运行/删除/克隆 | 查看待运行任务行 | 显示运行、删除、克隆按钮；不显示暂停、恢复 |
| TC-099-02 | 运行中状态→可见暂停/取消 | 查看运行中任务行 | 显示暂停、取消按钮；不显示运行、删除 |
| TC-099-03 | 已暂停状态→可见恢复/取消/重启 | 查看已暂停任务行 | 显示恢复、取消、重启按钮；不显示暂停 |
| TC-099-04 | 已完成状态→可见克隆/删除 | 查看已完成任务行 | 显示克隆、删除按钮；不显示运行/暂停/恢复/取消 |
| TC-099-05 | 已失败状态→可见重启/克隆/删除 | 查看已失败任务行 | 显示重启、克隆、删除按钮 |
| TC-099-06 | 已取消状态→可见重启/克隆/删除 | 查看已取消任务行 | 显示重启、克隆、删除按钮 |

### 3.2 状态流转操作

| # | 用例名称 | 步骤 | 预期 |
|---|---------|------|------|
| TC-099-07 | 暂停运行中任务 | 找到运行中任务→点暂停 | 状态变为"已暂停"，操作按钮更新 |
| TC-099-08 | 恢复已暂停任务 | 找到已暂停任务→点恢复 | 状态变为"运行中"，操作按钮更新 |
| TC-099-09 | 取消运行中任务 | 找到运行中任务→点取消→确认 | 状态变为"已取消" |
| TC-099-10 | 取消已暂停任务 | 找到已暂停任务→点取消→确认 | 状态变为"已取消" |
| TC-099-11 | 重启已失败任务 | 找到已失败任务→点重启 | 状态变为"运行中"或"排队中" |
| TC-099-12 | 重启已取消任务 | 找到已取消任务→点重启 | 状态变为"运行中"或"排队中" |

### 3.3 克隆功能

| # | 用例名称 | 步骤 | 预期 |
|---|---------|------|------|
| TC-099-13 | 克隆任务→配置继承 | 克隆一个已完成任务 | 新任务创建成功，配置与原任务一致，状态为待运行 |
| TC-099-14 | 克隆任务→名称带后缀 | 克隆任务 | 克隆的任务名称带"(副本)"或类似后缀 |
| TC-099-15 | 克隆任务→列表更新 | 克隆任务后 | 任务列表中出现新任务，总数+1 |

### 3.4 删除功能

| # | 用例名称 | 步骤 | 预期 |
|---|---------|------|------|
| TC-099-16 | 删除任务→二次确认 | 点击删除按钮 | 弹出确认弹窗 |
| TC-099-17 | 确认删除→任务移除 | 确认删除 | 任务从列表消失，统计数更新 |
| TC-099-18 | 取消删除→任务保留 | 弹窗中点取消 | 任务仍在列表中 |
| TC-099-19 | 运行中任务不可删除 | 查看运行中任务 | 删除按钮不可见或禁用 |

### 3.5 状态流转校验（非法跳转拦截）

| # | 用例名称 | 步骤 | 预期 |
|---|---------|------|------|
| TC-099-20 | 已完成任务不可暂停 | 查看已完成任务 | 暂停按钮不可见 |
| TC-099-21 | 待运行任务不可恢复 | 查看待运行任务 | 恢复按钮不可见 |

---

### `task-crud.spec.ts` — 需要修改的用例

| 原用例 | 改动说明 |
|-------|---------|
| should create a task using custom mode | 重写：流程改为 维度→类型（联动）→对象（对象库选择），去掉 `evalObject` 自由文本 和 `targetModel` 输入 |
| should create a task using template mode | 重写：验证模板预填维度/类型/对象，验证自动锁定逻辑 |
| should open create modal with 4-step wizard | 更新：Step 1 表单字段变化（新增评测维度、对象库选择器） |
| should navigate wizard steps back and forth | 更新：适配新表单结构 |
| should view task detail with tabs | 更新：详情中需验证维度和对象展示 |

### `task-lifecycle.spec.ts` — 需要修改的用例

| 原用例 | 改动说明 |
|-------|---------|
| should create a task and verify initial status | 重写：使用新的创建流程 |
| should cancel a pending/running task | 适配：使用新创建流程 + 验证新的取消确认交互 |
| should delete a task | 适配：使用新创建流程 |
| should clone a task | 扩充：验证克隆配置继承、状态重置 |

### `data-consistency.spec.ts` — 需要修改的用例

| 原用例 | 改动说明 |
|-------|---------|
| should show created task in list... | 重写：使用新创建流程，验证维度/对象列显示 |
| should update stats after creating a task | 适配：使用新创建流程 |

### `batch-operations.spec.ts` — 可能需要修改

| 原用例 | 改动说明 |
|-------|---------|
| batch cancel/delete | 验证新状态（已暂停）下的批量操作行为 |

### `error-handling.spec.ts` — 需要修改

| 原用例 | 改动说明 |
|-------|---------|
| should show validation when submitting... | 重写：新的必填字段校验（维度、对象） |

---

## 五、新增测试文件

| 文件 | 用途 |
|------|-----|
| `specs/preset-task.spec.ts` | 预置任务相关测试（#98） |
| `specs/eval-dimension-linkage.spec.ts` | 评测维度×类型联动的详细测试（#103 核心） |

---

## 测试用例汇总

| 分类 | 新增用例数 | 修改用例数 |
|------|----------|----------|
| #103 创建流程重构 | 21 | 5 |
| #98 预置任务 | 8 | 0 |
| #99 生命周期管理 | 21 | 4 |
| **合计** | **50** | **9** |

## 实施计划

1. **Phase 0（当前）**：本 issue — 规划用例清单 ✅
2. **Phase 1**：等 #103 合入后 → 重构 `tasks.page.ts` Page Object，实现维度联动测试
3. **Phase 2**：等 #98 合入后 → 新增 `preset-task.spec.ts`
4. **Phase 3**：等 #99 合入后 → 重构 `task-lifecycle.spec.ts`，新增状态流转测试
5. **Phase 4**：回归验证所有修改的用例，确保 CI 通过

## 关联 Issue
- #103 评测任务创建流程重构
- #98 算子评测预置任务
- #99 评测任务生命周期管理
